src/Mvc/Mvc.Core/src/Routing/UrlHelperFactory.cs

```diff
@@ -47,8 +47,8 @@ public IUrlHelper GetUrlHelper(ActionContext context)
                 return urlHelper;
             }
 
-            var endpointFeature = httpContext.Features.Get<IEndpointFeature>();
-            if (endpointFeature?.Endpoint != null)
+            var endpoint = httpContext.GetEndpoint();
+            if (endpoint != null)
```

Comment:
I dunno if I like this pattern yet.

```suggestion
            if (httpContext.GetEndpoint() is { } endpoint)
```